File: Python/underautomation/universal_robots/primary_interface/global_variables.py
import typing
from underautomation.universal_robots.primary_interface.global_variables_firmware_version import GlobalVariablesFirmwareVersion
from underautomation.universal_robots.common.global_variable import GlobalVariable
from underautomation.universal_robots.primary_interface.global_variables_event_args import GlobalVariablesEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import GlobalVariables as global_variables

class GlobalVariables:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = global_variables()
		else:
			self._instance = _internal
	def values_updated(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ValuesUpdated+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def list_updated(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ListUpdated+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def get_all(self) -> typing.List[GlobalVariable]:
		return [GlobalVariable(x) for x in self._instance.GetAll()]
	def get_by_name(self, name: str) -> GlobalVariable:
		return GlobalVariable(self._instance.GetByName(name))
	def __repr__(self):
		return self._instance.ToString()
	@property
	def firmware_version(self) -> GlobalVariablesFirmwareVersion:
		return GlobalVariablesFirmwareVersion(self._instance.FirmwareVersion)

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/GlobalVariables.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;
using System;

namespace PrimaryInterface {
	/// <summary>
	/// List of all global variables
	/// </summary>
	public class GlobalVariables {

		/// <summary>
		/// Indicates which decoder is used used to read variables according to firmware version
		/// </summary>
		public GlobalVariablesFirmwareVersion FirmwareVersion { get; }

		/// <summary>
		/// Returns a list of all variables declared in the robot
		/// </summary>
		public GlobalVariable[] GetAll()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Get a variable by its name. Null is returned if the variable doesn't exist
		/// </summary>
		/// <param name="name">Variable name, not case sensitive</param>
		public GlobalVariable GetByName(string name)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Event raised at 10Hz when variable values are updated
		/// </summary>
		public event EventHandler<GlobalVariablesEventArgs> ValuesUpdated;

		/// <summary>
		/// Event raised whan the variable list changed. For example, after a program starts
		/// </summary>
		public event EventHandler<GlobalVariablesEventArgs> ListUpdated;

		/// <summary>
		/// Return a string where each line contains type, name and variable value
		/// </summary>
		public override string ToString()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}
	}
}
